test/integration/protocol_integration_test.cc

```diff
@@ -747,6 +747,278 @@ TEST_P(DownstreamProtocolIntegrationTest, LargeRequestHeadersAccepted) {
   testLargeRequestHeaders(95, 96);
 }
 
+// Tests StopAllIterationAndBuffer. Verifies decode-headers-return-stop-all-filter calls decodeData
+// once after iteration is resumed.
+TEST_P(DownstreamProtocolIntegrationTest, testDecodeHeadersReturnsStopAll) {
+  config_helper_.addFilter(R"EOF(
+name: call-decodedata-once-filter
+)EOF");
+  config_helper_.addFilter(R"EOF(
+name: decode-headers-return-stop-all-filter
+)EOF");
+  config_helper_.addFilter(R"EOF(
+name: passthrough-filter
+)EOF");
+
+  initialize();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+
+  // Sends a request with headers and data.
+  const int count = 70;
```

Comment:
I'm seeing bits of this in a few places.  anything we could toss into utility functions?

Reply:
Fixed. Thanks!